### ISSUE TITLE
WASM out of bounds access, backend laggy

### ISSUE THREAD

**Issue by aliabbas299792** [aliabbas299792/radio#2]
For the WASM stuff:
```
The character encoding of the HTML document was not declared. The document will render with garbled text in some browser configurations if the document contains characters from outside the US-ASCII range. The character encoding of the page must be declared in the document or in the transfer protocol. radio.erewhon.xyz
Cannot enlarge memory, asked to go up to 2147684352 bytes, but the limit is 2147483648 bytes! main.js:4567:6
Cannot enlarge memory, asked to go up to 2147688448 bytes, but the limit is 2147483648 bytes! main.js:4567:6
==42==AddressSanitizer CHECK failed: /b/s/w/ir/k/install/emscripten/system/lib/compiler-rt/lib/sanitizer_common/sanitizer_allocator_secondary.h:101 "((IsAligned(map_beg, page_size_))) != (0)" (0x0, 0x0) main.js:1839:8
Cannot enlarge memory, asked to go up to 2147491840 bytes, but the limit is 2147483648 bytes! main.js:4567:6
Error: RuntimeError: index out of bounds index.js:224:13
Cannot enlarge memory, asked to go up to 2147688448 bytes, but the limit is 2147483648 bytes! main.js:4567:6
==42==AddressSanitizer CHECK failed: /b/s/w/ir/k/install/emscripten/system/lib/compiler-rt/lib/sanitizer_common/sanitizer_allocator_secondary.h:101 "((IsAligned(map_beg, page_size_))) != (0)" (0x0, 0x0) main.js:1839:8
Script terminated by timeout at:
__sanitizer::BlockingMutex::Lock()@https://radio.erewhon.xyz/main.wasm:wasm-function[796]:0x3f36c
__asan::AppendToErrorMessageBuffer(char const*)@https://radio.erewhon.xyz/main.wasm:wasm-function[440]:0x2f2d4
__sanitizer::SharedPrintfCodeNoBuffer(bool, char*, int, char const*, void*)@https://radio.erewhon.xyz/main.wasm:wasm-function[825]:0x40c46
__sanitizer::SharedPrintfCode(bool, char const*, void*)@https://radio.erewhon.xyz/main.wasm:wasm-function[824]:0x40ab9
__sanitizer::Report(char const*, ...)@https://radio.erewhon.xyz/main.wasm:wasm-function[826]:0x40c8c
__asan::AsanCheckFailed(char const*, int, char const*, unsigned long long, unsigned long long)@https://radio.erewhon.xyz/main.wasm:wasm-function[541]:0x323cb
__sanitizer::CheckFailed(char const*, int, char const*, unsigned long long, unsigned long long)@https://radio.erewhon.xyz/main.wasm:wasm-function[674]:0x3b768
__sanitizer::LargeMmapAllocator<__asan::AsanMapUnmapCallback, __sanitizer::LargeMmapAllocatorPtrArrayStatic, __sanitizer::LocalAddressSpaceView>::Allocate(__sanitizer::AllocatorStats*, unsigned long, unsigned long)@https://radio.erewhon.xyz/main.wasm:wasm-function[351]:0x29e3b
__sanitizer::CombinedAllocator<__sanitizer::SizeClassAllocator32<__asan::AP32<__sanitizer::LocalAddressSpaceView> >, __sanitizer::LargeMmapAllocatorPtrArrayStatic>::Allocate(__sanitizer::SizeClassAllocator32LocalCache<__sanitizer::SizeClassAllocator32<__asan::AP32<__sanitizer::LocalAddressSpaceView> > >*, unsigned long, unsigned long)@https://radio.erewhon.xyz/main.wasm:wasm-function[317]:0x27fec
__asan::Allocator::Allocate(unsigned long, unsigned long, __sanitizer::BufferedStackTrace*, __asan::AllocType, bool)@https://radio.erewhon.xyz/main.wasm:wasm-function[315]:0x27a91
__asan::asan_malloc(unsigned long, __sanitizer::BufferedStackTrace*)@https://radio.erewhon.xyz/main.wasm:wasm-function[314]:0x27878
malloc@https://radio.erewhon.xyz/main.wasm:wasm-function[490]:0x30cf6
createExportWrapper/<@https://radio.erewhon.xyz/main.js:1230:20
playMusic@https://radio.erewhon.xyz/assets/index.js:208:38
toggleAudio/player.audio_ws.onmessage@https://radio.erewhon.xyz/assets/index.js:311:20
main.wasm:258924:1
```
![image](https://user-images.githubusercontent.com/22846134/127567632-621a196f-7539-444c-9424-695341294a06.png)

And for the backend I've added a bunch of unbuffered logging, hopefully will see something of use show up soon

**Comment by aliabbas299792** [comment]
![image](https://user-images.githubusercontent.com/22846134/127776212-6473527c-f98e-4cb8-ae50-d5b58684db80.png)

**Comment by aliabbas299792** [comment]
The JS side was genuinely just forgetting to free the decoded buffer, fixed in ac1087149d080e6249c9ace0422db0dfc45090ff, still not sure about why the backend becomes laggy after like 2 days of use

**Comment by aliabbas299792** [comment]
Pretty sure all fixed with b50f55a648c3f4823e4f4665369984e44eca018a
